Build not finding modules from `node_modules`

```
$ yarn build
yarn build v0.17.6
$ rm -rf dist && BROCCOLI_ENV=dist broccoli build dist
Build project: @glimmer/di
Build env: dist
Build path: /wouldnt/you/like/to/know/glimmer-di
/src/container.ts(4,28): Cannot find module 'glimmer-util'.
/src/registry.ts(1,28): Cannot find module 'glimmer-util'.
✨  Done in 3.31s.
```

Probably because we're not including `node_modules` when building:

```js
let tsinclude = [
  'src/**',
  'lib.*.d.ts'
];

// ...

let ts = funnel(mergeTrees([libs, projectPath]), {
  include: tsinclude,
  annotation: 'raw source'
});
```

--- a/index.js
+++ b/index.js
@@ -29,15 +29,7 @@ module.exports = function(options) {
 
   let trees = [];
 
-  let tsinclude = [
-    'src/**',
-    'lib.*.d.ts'
-  ];
-
   if (env === 'tests') {
-    tsinclude.push('test/**');
-    tsinclude.push('node_modules/@types/**/*.ts');
-
     trees.push(funnelLib('loader.js', {
       include: ['loader.js'],
       annotation: 'loader.js'
@@ -73,9 +65,9 @@ module.exports = function(options) {
       annotation: 'vendor.js'
     }));
 
-    trees.push(compileTS('tsconfig.tests.json', projectPath, tsinclude));
+    trees.push(compileTS('tsconfig.tests.json', projectPath));
   } else {
-    let es2017ModulesAndTypes = compileTS('tsconfig.json', projectPath, tsinclude);
+    let es2017ModulesAndTypes = compileTS('tsconfig.json', projectPath);
     let types = selectTypesFromTree(es2017ModulesAndTypes);
     let es2017Modules = filterTypescriptFromTree(es2017ModulesAndTypes);
     let es5Modules = toES5(es2017Modules, { sourceMap: 'inline' });
@@ -113,7 +105,7 @@ module.exports = function(options) {
   return mergeTrees(trees);
 };
 
-function compileTS(tsconfigFile, projectPath, tsinclude) {
+function compileTS(tsconfigFile, projectPath) {
   let tsconfig = JSON.parse(fs.readFileSync(tsconfigFile));
 
   if (tsconfig.compilerOptions.outFile) {
@@ -129,7 +121,6 @@ function compileTS(tsconfigFile, projectPath, tsinclude) {
   });
 
   let ts = funnel(mergeTrees([libs, projectPath]), {
-    include: tsinclude,
     annotation: 'raw source'
   });
 

--- a/index.js
+++ b/index.js
@@ -1,20 +1,18 @@
 "use strict";
 
-const typescript = require('broccoli-typescript-compiler').typescript;
-const mergeTrees = require('broccoli-merge-trees');
 const assert = require('assert');
-const funnel = require('broccoli-funnel');
+const compileTypescript = require('./lib/compile-typescript');
 const concat = require('broccoli-concat');
-const fs = require('fs');
-const replace = require('broccoli-string-replace');
-const writeFile = require('broccoli-file-creator');
-const findLib = require('./lib/find-lib');
+const funnel = require('broccoli-funnel');
 const funnelLib = require('./lib/funnel-lib');
 const getPackageName = require('./lib/get-package-name');
+const helpers = require('./lib/generate-helpers');
+const mergeTrees = require('broccoli-merge-trees');
+const replace = require('broccoli-string-replace');
+const toES5 = require('./lib/to-es5');
 const toNamedAmd = require('./lib/to-named-amd');
 const toNamedCommonJs = require('./lib/to-named-common-js');
-const toES5 = require('./lib/to-es5');
-const helpers = require('./lib/generate-helpers');
+const writeFile = require('broccoli-file-creator');
 
 module.exports = function(options) {
   options = options || {};
@@ -65,9 +63,9 @@ module.exports = function(options) {
       annotation: 'vendor.js'
     }));
 
-    trees.push(compileTS('tsconfig.tests.json', projectPath));
+    trees.push(compileTypescript('tsconfig.tests.json', projectPath));
   } else {
-    let es2017ModulesAndTypes = compileTS('tsconfig.json', projectPath);
+    let es2017ModulesAndTypes = compileTypescript('tsconfig.json', projectPath);
     let types = selectTypesFromTree(es2017ModulesAndTypes);
     let es2017Modules = filterTypescriptFromTree(es2017ModulesAndTypes);
     let es5Modules = toES5(es2017Modules, { sourceMap: 'inline' });
@@ -105,43 +103,10 @@ module.exports = function(options) {
   return mergeTrees(trees);
 };
 
-function compileTS(tsconfigFile, projectPath) {
-  let tsconfig = JSON.parse(fs.readFileSync(tsconfigFile));
-
-  if (tsconfig.compilerOptions.outFile) {
-    tsconfig.compilerOptions.outFile = removeFirstPathSegment(tsconfig.compilerOptions.outFile);
-  }
-
-  if (tsconfig.compilerOptions.outDir) {
-    tsconfig.compilerOptions.outDir = removeFirstPathSegment(tsconfig.compilerOptions.outDir);
-  }
-
-  let libs = funnel(findLib('typescript'), {
-    include: ['lib.*.d.ts']
-  });
-
-  let ts = funnel(mergeTrees([libs, projectPath]), {
-    annotation: 'raw source'
-  });
-
-  let compiledTS = typescript(ts, {
-    tsconfig,
-    annotation: 'compiled source'
-  });
-
-  return compiledTS;
-}
-
 function selectTypesFromTree(tree) {
   return funnel(tree, { include: ['**/*.d.ts'] });
 }
 
 function filterTypescriptFromTree(tree) {
   return funnel(tree, { exclude: ['**/*.ts'] });
-}
-
-function removeFirstPathSegment(path) {
-  let parts = path.split('\/');
-  parts.shift();
-  return parts.join('\/');
 }

--- a/lib/compile-typescript.js
+++ b/lib/compile-typescript.js
@@ -1,0 +1,38 @@
+"use strict";
+
+const findLib = require('./find-lib');
+const fs = require('fs');
+const funnel = require('broccoli-funnel');
+const mergeTrees = require('broccoli-merge-trees');
+const typescript = require('broccoli-typescript-compiler').typescript;
+
+module.exports = function compileTypescript(tsconfigFile, projectPath) {
+  let tsconfig = JSON.parse(fs.readFileSync(tsconfigFile));
+
+  let libs = funnel(findLib('typescript'), {
+    include: ['lib.*.d.ts']
+  });
+
+  let ts = funnel(mergeTrees([libs, projectPath]), {
+    annotation: 'raw source'
+  });
+
+  if (tsconfig.compilerOptions.outFile) {
+    tsconfig.compilerOptions.outFile = removeFirstPathSegment(tsconfig.compilerOptions.outFile);
+  }
+
+  if (tsconfig.compilerOptions.outDir) {
+    tsconfig.compilerOptions.outDir = removeFirstPathSegment(tsconfig.compilerOptions.outDir);
+  }
+
+  return typescript(ts, {
+    tsconfig,
+    annotation: 'compiled source'
+  });
+};
+
+function removeFirstPathSegment(path) {
+  let parts = path.split('\/');
+  parts.shift();
+  return parts.join('\/');
+}

--- a/lib/funnel-lib.js
+++ b/lib/funnel-lib.js
@@ -9,7 +9,6 @@ module.exports = function funnelLib(name) {
     libPath = arguments[1];
     options = arguments[2];
   } else {
-    libPath = '.';
     options = arguments[1];
   }
   return funnel(findLib(name, libPath), options);

--- a/test/compile-typescript-test.js
+++ b/test/compile-typescript-test.js
@@ -1,0 +1,110 @@
+"use strict";
+
+const broccoli = require('broccoli');
+const expect = require('chai').expect;
+const compileTypescript = require('../lib/compile-typescript');
+const fixturify = require('fixturify');
+const fs = require('fs');
+const funnel = require('broccoli-funnel');
+const path = require('path');
+const rimrafSync = require('rimraf').sync;
+const walkSync = require('walk-sync');
+
+const TMP_DIR = path.join(__dirname, '../tmp/compile-typescript');
+
+let builder;
+
+describe('compile-typescript', function() {
+  afterEach(function() {
+    teardown();
+  });
+
+  describe('when no outDir or outFile are specified', function() {
+    beforeEach(function() {
+      setup({
+        compilerOptions: {},
+        include: [
+          'src/**/*.ts'
+        ]
+      });
+    });
+
+    it('compiles typescript', function() {
+      let tsconfigFile = path.join(TMP_DIR, 'tsconfig.json');
+      let projectPath = TMP_DIR;
+      let compiledTypescript = compileTypescript(tsconfigFile, projectPath);
+
+      builder = new broccoli.Builder(compiledTypescript);
+
+      return builder.build().then(() => {
+        expect(walkSync(builder.outputPath)).to.eql(['src/', 'src/foo.js']);
+      });
+    });
+  });
+
+  describe('when outDir is specified', function() {
+    beforeEach(function() {
+      setup({
+        compilerOptions: {
+          outDir: 'dist/lib'
+        },
+        include: [
+          'src/**/*.ts'
+        ]
+      });
+    });
+
+    it('compiles typescript into shifted outDir', function() {
+      let tsconfigFile = path.join(TMP_DIR, 'tsconfig.json');
+      let projectPath = TMP_DIR;
+      let compiledTypescript = compileTypescript(tsconfigFile, projectPath);
+
+      builder = new broccoli.Builder(compiledTypescript);
+
+      return builder.build().then(() => {
+        expect(walkSync(builder.outputPath)).to.eql(['lib/', 'lib/foo.js']);
+      });
+    });
+  });
+
+  describe('when outFile is specified', function() {
+    beforeEach(function() {
+      setup({
+        compilerOptions: {
+          outFile: 'dist/lib/foo.js'
+        },
+        include: [
+          'src/**/*.ts'
+        ]
+      });
+    });
+
+    it('compiles typescript into shifted outFile', function() {
+      let tsconfigFile = path.join(TMP_DIR, 'tsconfig.json');
+      let projectPath = TMP_DIR;
+      let compiledTypescript = compileTypescript(tsconfigFile, projectPath);
+
+      builder = new broccoli.Builder(compiledTypescript);
+
+      return builder.build().then(() => {
+        expect(walkSync(builder.outputPath)).to.eql(['lib/', 'lib/foo.js']);
+      });
+    });
+  });
+});
+
+function setup(tsconfig) {
+  fixturify.writeSync(TMP_DIR, {
+    'tsconfig.json': JSON.stringify(tsconfig),
+    'src': {
+      'foo.ts': ''
+    }
+  });
+}
+
+function teardown() {
+  rimrafSync(TMP_DIR);
+  if (builder) {
+    return builder.cleanup();
+  }
+}

